### ISSUE TITLE
Wagtail 3.0 error? `TabbedInterface.bind_to_model(model) must be called before get_bound_panel`

### ISSUE THREAD

**Issue by janbaykara** [jazzband/wagtailmenus#421]
When opening a main menu / flat menu edit page, an uncaught error.

### Environment

Runing python 3.9, Django 4, Wagtail 3.0

### Frontend report

```
<h1>ImproperlyConfigured
       at /admin/wagtailmenus/flatmenu/edit/3/</h1>
<pre class="exception_value">TabbedInterface.bind_to_model(model) must be called before get_bound_panel</pre>

Request Method: | GET
-- | --
http://localhost:8000/admin/wagtailmenus/flatmenu/edit/3/
4.0.4
ImproperlyConfigured
TabbedInterface.bind_to_model(model) must be called before get_bound_panel
/home/vscode/.cache/pypoetry/virtualenvs/app-xS3fZVNL-py3.9/lib/python3.9/site-packages/wagtail/admin/panels.py, line 235, in get_bound_panel
/home/vscode/.cache/pypoetry/virtualenvs/app-xS3fZVNL-py3.9/bin/python
```

### Full stack trace

```
Traceback (most recent call last):
File "D:\Users\ani\Documents\venv\lib\site-packages\django\core\handlers\exception.py", line 47, in inner
response = get_response(request)
File "D:\Users\ani\Documents\venv\lib\site-packages\django\core\handlers\base.py", line 181, in _get_response
response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "D:\Users\ani\Documents\venv\lib\site-packages\django\views\decorators\cache.py", line 44, in wrapped_view_func
response = view_func(request, *args, **kwargs)
File "D:\Users\ani\Documents\venv\lib\site-packages\wagtail\admin\urls_init.py", line 161, in wrapper
return view_func(request, *args, **kwargs)
File "D:\Users\ani\Documents\venv\lib\site-packages\wagtail\admin\auth.py", line 182, in decorated_view
response = view_func(request, *args, **kwargs)
File "D:\Users\ani\Documents\venv\lib\site-packages\wagtail\contrib\modeladmin\options.py", line 445, in edit_view
return view_class.as_view(**kwargs)(request)
File "D:\Users\ani\Documents\venv\lib\site-packages\django\views\generic\base.py", line 64, in view
self.setup(request, *args, **kwargs)
File "D:\Users\ani\Documents\venv\lib\site-packages\wagtail\contrib\modeladmin\views.py", line 161, in setup
self.edit_handler = self.get_edit_handler()
File "D:\Users\ani\Documents\venv\lib\site-packages\wagtailmenus\views.py", line 59, in get_edit_handler
return edit_handler.bind_to(model=self.model)
File "D:\Users\ani\Documents\venv\lib\site-packages\wagtail\admin\panels.py", line 227, in bind_to
return self.get_bound_panel(instance=instance, request=request, form=form)
File "D:\Users\ani\Documents\venv\lib\site-packages\wagtail\admin\panels.py", line 237, in get_bound_panel
% type(self).name

Exception Type: ImproperlyConfigured at /gavekal-wagtail-admin/wagtailmenus/mainmenu/edit/2/
Exception Value: TabbedInterface.bind_to_model(model) must be called before get_bound_panel
```

### Relevant code snippets

https://github.com/jazzband/wagtailmenus/blob/8be2790c01688dd3c9ef7298bdebe94575c0a76a/wagtailmenus/views.py#L50-L60

**Comment by aliceni81** [comment]
Here is the traceback:

Traceback (most recent call last):
  File "D:\Users\ani\Documents\venv\lib\site-packages\django\core\handlers\exception.py", line 47, in inner
    response = get_response(request)
  File "D:\Users\ani\Documents\venv\lib\site-packages\django\core\handlers\base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "D:\Users\ani\Documents\venv\lib\site-packages\django\views\decorators\cache.py", line 44, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "D:\Users\ani\Documents\venv\lib\site-packages\wagtail\admin\urls\__init__.py", line 161, in wrapper
    return view_func(request, *args, **kwargs)
  File "D:\Users\ani\Documents\venv\lib\site-packages\wagtail\admin\auth.py", line 182, in decorated_view
    response = view_func(request, *args, **kwargs)
  File "D:\Users\ani\Documents\venv\lib\site-packages\wagtail\contrib\modeladmin\options.py", line 445, in edit_view
    return view_class.as_view(**kwargs)(request)
  File "D:\Users\ani\Documents\venv\lib\site-packages\django\views\generic\base.py", line 64, in view
    self.setup(request, *args, **kwargs)
  File "D:\Users\ani\Documents\venv\lib\site-packages\wagtail\contrib\modeladmin\views.py", line 161, in setup
    self.edit_handler = self.get_edit_handler()
  File "D:\Users\ani\Documents\venv\lib\site-packages\wagtailmenus\views.py", line 59, in get_edit_handler
    return edit_handler.bind_to(model=self.model)
  File "D:\Users\ani\Documents\venv\lib\site-packages\wagtail\admin\panels.py", line 227, in bind_to
    return self.get_bound_panel(instance=instance, request=request, form=form)
  File "D:\Users\ani\Documents\venv\lib\site-packages\wagtail\admin\panels.py", line 237, in get_bound_panel
    % type(self).__name__

Exception Type: ImproperlyConfigured at /gavekal-wagtail-admin/wagtailmenus/mainmenu/edit/2/
Exception Value: TabbedInterface.bind_to_model(model) must be called before get_bound_panel

**Comment by MrCordeiro** [comment]
Hi, wagtailmenus is still not compatible with Wagtail 3.0. This is what's causing the error.

**Comment by basitiyidir** [comment]
I want to start a new project with Wagtail 3.
When will this bug be fixed.

how can i fix it temporarily

Thank you.

**Comment by janbaykara** [comment]
Can someone who's familiar with Wagtail's panel system comment here with roughly what the problem seems to be? I poked around the code for 5 minutes but it didn't occur. Happy to put up a fix but I can't invest the debugging time right now.

**Comment by basitiyidir** [comment]
> @amajai nice sleuthing. Do you want to submit this as a PR to close this issue?

I'm waiting for someone to PR.

**Comment by amajai** [comment]
Alright, I'll make a PR.

**Comment by amajai** [comment]
@janbaykara I was actually going to post PR later today. Had to go over the code and documentation to make proper changes during my free time. Sorry that it took long.

**Comment by amajai** [comment]
Temporary fix to wagtail 3.0 for now:
pip install -e git+https://github.com/amajai/wagtailmenus.git@resolve-wagtailmenus-compatibility-issue-wagtail-3.0#egg=wagtail

**Comment by basitiyidir** [comment]
> Temporary fix to wagtail 3.0 for now until wagtailmenus gets updated:
> pip install -e git+https://github.com/amajai/wagtailmenus.git@resolve-wagtailmenus-compatibility-issue-wagtail-3.0#egg=wagtail

Thank you.
But the following error...

`pip._vendor.resolvelib.resolvers.InconsistentCandidate: Provided candidate LinkCandidate('https://files.pythonhosted.org/packages/cf/82/d16a3836856383d2f34686226b927cf076f01da4be49654717c923279d6e/wagtail-3.0-py3-none-any.whl#sha256=4a844d2d1cf5cbc9c48344c6ae9cc5c9f0d1134cd45cd2c484afdedc1164043a (from https://pypi.org/simple/wagtail/) (requires-python:>=3.7)') does not satisfy SpecifierRequirement('wagtail==3.0'), UnsatisfiableRequirement('wagtail')`

**Comment by amajai** [comment]
@basitiyidir Try updating your python version to 3.7 or above.

**Comment by basitiyidir** [comment]
> @basitiyidir Try updating your python version to 3.7 or above.

I am using docker and the python version is python:3.10.2.

`FROM python:3.10.2-slim

WORKDIR /code

ENV PYTHONDONTWRITEBYTECODE 1

ENV PYTHONUNBUFFERED 1


RUN apt-get update && apt-get install -qq -y \
    build-essential libpq-dev libffi-dev --no-install-recommends git \
    python3-dev python3-setuptools

RUN pip install --upgrade pip
COPY ./req_dev.txt /code/req_dev.txt
RUN pip install -r req_dev.txt
COPY ./entrypoint.sh /code/
COPY . /code`

**Comment by MrCordeiro** [comment]
Closed by #424 

Thanks everyone for your help! ✌️

Please be aware that release to PyPi is still being blocked by #419 and #409. In the meanwhile, you can get the latest version on GitHub directly.
